lib/web_ui/dev/browser_roller.dart

```diff
@@ -0,0 +1,318 @@
+    } else {
+      // Clean-up
+      vprint('\nDeleting temporary directory: ${_rollDir.path}');
+      await _rollDir.delete(recursive: true);
+      print('\nDone.\n');
+    }
+  }
+
+  // Returns the contents for the CIPD config required to publish a new chromium package.
+  String _getCipdChromiumConfig({
+    required String platform,
+    required String version,
+    required String build,
+    required String root,
+  }) {
+    return '''
+package: flutter_internal/browsers/chrome/$platform-amd64
+description: Chromium $version (build $build) used for testing
+preserve_writable: true
+root: $root
+data:
+  - dir: .
+''';
+  }
+
+  // Returns the contents for the CIPD config required to publish a new chromedriver package.
+  String _getCipdChromedriverConfig({
+    required String platform,
+    required String version,
+    required String build,
+    required String root,
+  }) {
+    return '''
+package: flutter_internal/browser-drivers/chrome/$platform-amd64
+description: Chromedriver for Chromium $version (build $build) used for testing
+preserve_writable: true
+root: $root
+data:
+  - dir: .
+''';
+  }
+
+  // Download a file from the internet, and put it in a temporary location.
+  Future<io.File> _downloadTemporaryFile(String url) async {
+    // Use the hash of the Url to temporarily store a file under tmp
+    final io.File downloadedFile = io.File(path.join(
+        io.Directory.systemTemp.path,
+        'download_' + url.hashCode.toRadixString(16),
+      ));
+    vprint('  Downloading [$url] into [${downloadedFile.path}]');
+    final StreamedResponse download = await _client.send(
+      Request('GET', Uri.parse(url)),
+    );
+    await download.stream.pipe(downloadedFile.openWrite());
+    return downloadedFile;
+  }
+
+  // Unzips a `file` into a `destination` Directory (must exist).
+  Future<void> _unzipAndDeleteFile(io.File zipFile, io.Directory destination) async {
+    vprint('  Unzipping [${zipFile.path}] into [$destination]');
+    await runProcess('unzip', <String>[
+      if (!verbose) ...<String>[
+        '-q',
+      ],
+      zipFile.path,
+      '-d',
+      destination.path,
+    ]);
+    vprint('  Deleting [${zipFile.path}]');
+    await zipFile.delete();
+  }
+
+  // Write String `contents` to a file in `path`.
+  //
+  // This is used to write CIPD config files to disk.
+  Future<io.File> _writeFile(String path, String contents) async {
+    vprint('  Writing file [$path]');
+    final io.File file = io.File(path, );
+    await file.writeAsString(contents);
+    return file;
+  }
+
+  // Locate the first subdirectory that contains more than one file under `root`.
+  // (or one ".app" bundle for mac)
+  //
+  // When uncompressing files, unzip might create some extra directories, but it
+  // seems that our scripts want our CIPD packages to contain everything in the root.
+  Future<io.Directory?> _locateContentRoot(io.Directory root) async {
```

Comment:
This won't be necessary given the convention above

Reply:
Removed

Reply:
Blah, I don't need this for the chromium rolls, but DO for the chromedriver ones, yiiikes